src/OpenSage.Game/Logic/Orders/OrderProcessor.cs

```diff
@@ -61,30 +91,81 @@ public void Process(IEnumerable<Order> orders)
-                            .Select(x => (int) x.Value.ObjectId)
-                            .Select(_game.Scene3D.GameObjects.GetObjectById)
-                            .ToArray();
+                            _game.Selection.SetSelectedObjects(player, objectIds);
+                        }
+                        catch (System.Exception e)
+                        {
+                            logger.Error(e, "Error while setting selection");
+                        }
 
-                        _game.Selection.SetSelectedObjects(player, objectIds);
                         break;
 
                     case OrderType.ClearSelection:
                         _game.Selection.ClearSelectedObjects(player);
                         break;
 
                     case OrderType.SetRallyPoint:
-                          var objIds = order.Arguments.Skip(1)
-                            .Select(x => (int) x.Value.ObjectId)
-                            .Select(_game.Scene3D.GameObjects.GetObjectById)
-                            .ToArray();
-                        _game.Selection.SetRallyPointForSelectedObjects(player, objIds, new Vector3());
+                        try
+                        {
+                            if(order.Arguments.Count == 2)
+                            {
+                                var objId = order.Arguments[0].Value.ObjectId;
+                                var obj = _game.Scene3D.GameObjects.GetObjectById((int)objId);
+
+                                var rallyPoint = order.Arguments[1].Value.Position;
+                                obj.RallyPoint = rallyPoint;
+                            }
+                            else
+                            {
+                                var objIds = order.Arguments.Skip(1)
+                                    .Select(x => (int) x.Value.ObjectId)
+                                    .Select(_game.Scene3D.GameObjects.GetObjectById)
+                                    .ToArray();
+                                _game.Selection.SetRallyPointForSelectedObjects(player, objIds, new Vector3());
+                            }
+                            
+
+                        }
+                        catch (System.Exception e)
+                        {
+                            logger.Error(e, "Error while setting rallypoint");
+                        }
                         break;
 
                     case OrderType.Unknown27:
                         _game.EndGame();
                         break;
+
+                    //case OrderType.ChooseGeneralPromotion:
+                        //gla:
+                        //tier 1:
+                        //34, 35, 36
+
+                        //usa:
+                        //tier1:
+                        //12, 13, 14
+                    //    break;
+
+                    default:
+                        var args = new StringBuilder();
```

Comment:
I think you can replace all of this with:
```csharp
var args = string.Join(", ", order.Arguments.Select(argument => argument.ToString()));
```

Reply:
Done